app/plugins/hmac-authentication.md

```diff
@@ -115,45 +133,189 @@ the order they appear.
   lowercased header name followed with an ASCII colon `:` and an
   ASCII space ` `.
 
-2. If the header name is `request-line` then appened the HTTP
+2. If the header name is `request-line` then append the HTTP
   request line, otherwise append the header value.
 
 3. If value is not the last value then append an ASCII newline `\n`.
   The string MUST NOT include a trailing ASCII newline.
 
 ### Clock Skew
 
-The HMAC Authentication plugin also implements a clock skew check [as described in the specification](https://tools.ietf.org/html/draft-cavage-http-signatures-00#section-3.4) to prevent replay attacks. By default, a minimum lag of 300s in either direction (past/future) is allowed. Any request with a higher or lower date value will be rejected. The length of the clock skew can be edited through the plugin's configuration by setting `clock_skew` property (`config.clock_skew` POST parameters).
+The HMAC Authentication plugin also implements a clock skew check as described
+in the [specification][clock-skew] to prevent replay attacks. By default,
+a minimum lag of 300s in either direction (past/future) is allowed. Any request
+with a higher or lower date value will be rejected. The length of the clock
+skew can be edited through the plugin's configuration by setting `clock_skew`
+property (`config.clock_skew` POST parameters).
+
+The server and requesting client should be synchronized with NTP and a valid
+date (using GMT format) should be sent with either the `X-Date `or `Date`
+header.
+
+### Body Validation
+  
+User can set `config.validate_request_body` as `true` to validate the request 
+body. If it's enabled and if the client sends a `Digest` header in the request,
+plugin will calculate the `SHA-256` HMAC digest of request body and match it
+against the value of `Digest` header. Digest header needs to be in following
```

Comment:
value of *the* `Digest` header. *The* `Digest` header needs to *follow the* following format: